app/views/api_views.py

```diff
@@ -20,6 +20,34 @@ def autocomplete_name(request):
         return HttpResponse('', content_type="application/json")
 
 
+@login_required(login_url='/login')
+@require_GET
+def donor_names(request):
+    try:
+        donor_names = [donor.donor_name for donor in Donor.objects.all()]
+        return JsonResponse({'donorNames': donor_names},
+                            content_type="application/json")
+    except:
+        return HttpResponse('', content_type="application/json")
+
+
+@login_required(login_url='/login')
+@require_GET
+def device_names(request):
+    try:
+        device_objs = []
+        for device in ItemDevice.objects.all():
+            if device.dtype is not None:
+                deviceInfo = device.dtype.device_type + \
```

Comment:
please use f-string for readability. https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/